### PR TITLE
Update 03-index-slice-subset.md

### PR DESCRIPTION
**Pull request by Isaac-Jones**
Updated lesson to comply with Issue 436

Changed "Lesson 3" reference to the correct "Lesson 4" reference, along with a link to the lesson.

### CODE REVIEW

File: _episodes/03-index-slice-subset.md [hunk 1]
```diff
@@ -526,7 +526,7 @@ surveys_df[pd.isnull(surveys_df).any(axis=1)]
 {: .language-python}
 
 Note that the `weight` column of our DataFrame contains many `null` or `NaN`
-values. We will explore ways of dealing with this in Lesson 03.
+values. We will explore ways of dealing with this in [Lesson 04](https://datacarpentry.org/python-ecology-lesson/04-data-types-and-format/index.html).
```
Comment (maxim-belkin) [inline _episodes/03-index-slice-subset.md:529]: 1. Here we're talking about the next *episode* (this whole repository is a single lesson, which is comprised of several episodes), so I suggest to tweak the wording and say something along the lines of `...explore ways of dealing with this in the next episode on [Data Types and Formats](...)`.
2. Let's use Carpentries' recommended way of linking to other episodes, which is described here: https://carpentries.github.io/lesson-example/04-formatting/index.html#locations-and-names

**Review by maxim-belkin — CHANGES_REQUESTED** [review]
Your changes make perfect sense, @Isaac-Jones, so I've got only a couple of comments.
Thank you for your contribution to the lesson!!! 👍

File: _episodes/03-index-slice-subset.md [hunk 2]
```diff
@@ -526,7 +526,7 @@ surveys_df[pd.isnull(surveys_df).any(axis=1)]
 {: .language-python}
 
 Note that the `weight` column of our DataFrame contains many `null` or `NaN`
-values. We will explore ways of dealing with this in Lesson 03.
+values. We will explore ways of dealing with this in the next episode on [Data Types and Formats]({{ page.root }}{% _episodes/04-data-types-and-format.md %}).
```
Comment (maxim-belkin) [inline _episodes/03-index-slice-subset.md:529]: `{% link _episodes/04-data-types-and-format.md %}`

**Comment by Isaac-Jones** [comment]
I am having trouble with the linking format... Is this what it is supposed to look like?

File: _episodes/03-index-slice-subset.md (hunk 2 above)
Comment (Isaac-Jones) [reply _episodes/03-index-slice-subset.md]: got it, thanks!

**Review by maxim-belkin — APPROVED** [review]
👍 Thank you, Isaac! Yes, the syntax for linking to other episodes is not as straightforward as it could've been but you figured it out, so great job!

**Pull request merged**